src/snowflake/connector/connection_diagnostic.py

```diff
@@ -0,0 +1,578 @@
+            connect_creds: str = ""
+            conn = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            conn.settimeout(timeout)
+            if self.proxy_host is not None:
+                proxy_addr = (self.proxy_host, self.proxy_port)
+                if self.proxy_user is not None:
+                    proxy_auth = f"{self.proxy_user}:{self.proxy_password}"
+                    proxy_auth = proxy_auth.encode("utf-8")
+                    credentials = base64.b64encode(proxy_auth).decode().strip("\n")
+                    connect_creds = f"Proxy-Authorization: Basic {credentials}\r\n"
+                conn.connect(proxy_addr)
+            else:
+                conn.connect((host, int(port)))
+
+            if port == 443:
+                if self.proxy_host is not None:
+                    connect = f"CONNECT {host}:{port} HTTP/1.1\r\n{connect_creds}"
+                    connect = f"{connect}Host: {host}\r\n\r\n"
+                    conn.send(str.encode(connect))
+                    conn.recv(4096).decode("utf-8")
+
+                context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+                sock = context.wrap_socket(conn, server_hostname=host)
+                certificate = ssl.DER_cert_to_PEM_cert(sock.getpeercert(True))
+                conn.close()
+                return certificate
+            else:
+                if self.proxy_host is not None:
+                    connect = f"CONNECT {host}:{port} HTTP/1.1\r\n{connect_creds}\r\n\r\n"
+                else:
+                    connect = f"GET / HTTP/1.1\r\nHost: {host}\r\nConnection: close\r\n\r\n"
+
+                conn.send(str.encode(connect))
+                response = conn.recv(4096).decode("utf-8")
+                conn.close()
+
+            if response is not None:
+                good_responses = "(200|301|cloudfront)"
+                if not re.search(good_responses, response):
+                    self.__append_message(host_type, f"{host}:{port}: URL Check: Failed: {response}")
+                    return "FAILED"
+            self.__append_message(host_type, f"{host}:{port}: URL Check: Connected Successfully")
+            return "SUCCESS"
+        except ssl.SSLError as e:
+            if 'WRONG_VERSION_NUMBER' in str(e):
+                self.__append_message(host_type, f"{host}:{port}: URL Check: Failed: Proxy Auth Error: {e}")
+            return "FAILED"
+        except Exception as e:
+            self.__append_message(host_type, f"{host}:{port}: URL Check: Failed: Unknown Exception: {e}")
+            return "FAILED"
+
+        self.__append_message(host_type, f"{host}:{port}: URL Check: Connected Successfully")
+        return "SUCCESS"
+
+    def run_post_test(self) -> None:
+        if self.full_connection_diag_whitelist_path is None:
+            if self.cursor is not None:
+                try:
+                    results = self.cursor.execute(self.whitelist_sql).fetchall()[0][0]
```

Comment:
Flipping this flag would hide these SQL commands from showing up in history by default
```suggestion
                    results = self.cursor.execute(self.whitelist_sql, _is_internal=True).fetchall()[0][0]
```